group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorShard.java

```diff
@@ -140,7 +159,12 @@ public GroupCoordinatorShard build() {
                 throw new IllegalArgumentException("Time must be set.");
             if (timer == null)
                 throw new IllegalArgumentException("Timer must be set.");
+            if (coordinatorMetrics == null)
+                throw new IllegalArgumentException("CoordinatorMetricsShard must be set.");
```

Comment:
I noticed we check the metrics shard to ensure it is a GroupCoordinatorMetrics shard in another file. Should we check the type sooner? Should we also check this is the type GroupCoordinatorMetrics (also should the error not include shard here)?